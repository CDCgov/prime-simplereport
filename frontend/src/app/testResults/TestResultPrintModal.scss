.sr-result-print-buttons {
  display: flex;
  justify-content: flex-end;

  & button {
    margin: 0 0 0 1.5rem;
  }
}

@media print {
  body {
    .dont-print {
      display: none !important;
    }

    .sr-test-results-modal-content {
      transform: none;
      max-height: none;
      padding: 0;
    }
  }
}

.sr-removed-result::before {
  position: absolute;
  font-size: 4rem;
  opacity: 50%;
  color: #666;
  transform: rotate(-30deg) translate(-8rem, 12rem);
  outline: 0.2rem solid #888;
  -webkit-text-stroke: 1px #fff;
  content: "MARKED AS ERROR";
  background: transparent;
}

.sr-test-results-modal-content {
  font-size: 80% !important;
  background: #fff;
  margin-top: 2%;
  inset: 3em auto auto auto;
  overflow: auto;
  max-height: 90vh;
  width: 7.6in;
  padding: 0.4in;
}

.sr-test-results-modal-overlay {
  display: flex;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.8);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 1000;
}

.sr-test-result-report {
  header {
    & > .sr-print-logo {
      float: right;
      width: 200px;
    }

    h1 {
      padding: 0;
      margin: 0;
    }
  }

  main {
    display: flex;
    flex-wrap: wrap;
  }

  footer {
    p {
      margin-bottom: 0;
    }
  }
}

.sr-result-section {
  width: 100%;
  border: 1px solid #767676;
  margin: 0.25rem 0;

  h2 {
    font-size: 100%;
    font-weight: normal;
    color: #fcfcfc;
    background: #6c6c6c;
    margin: 0;
    padding: 0.5rem;

    /* allow dark background on header */
    -webkit-print-color-adjust: exact;
  }

  h3 {
    margin: 0.75rem 1rem;
  }

  ul.sr-multi-column {
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    max-height: 14ex;
  }

  p {
    margin: 0.75rem 1rem;
  }
}

.sr-result-patient-details,
.sr-result-facility-details {
  width: 49%;
}

.sr-result-facility-details {
  margin-left: 2%;
}

.sr-result-test-details {
  .sr-details-list {
    li {
      b {
        width: unset !important;
        min-width: 15rem;
        max-width: 15rem;
      }
    }
  }
}

.sr-details-list {
  list-style-type: none;
  margin-left: 0.5rem;
  padding: 0;

  li {
    margin: 0.5rem;
    display: flex;
    flex-direction: row;

    b {
      width: 8rem;
      flex-shrink: 0;
    }

    .sr-result-facility-details-address span {
      display: block;
    }

    div {
      margin-left: 0.5rem;
    }
  }

  div:empty::after {
    content: ".";
    visibility: hidden;
  }
}
