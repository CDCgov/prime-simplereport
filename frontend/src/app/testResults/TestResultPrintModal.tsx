import React from "react";
import { gql } from "@apollo/client";
import Modal from "react-modal";
import classnames from "classnames";
import { Trans, useTranslation } from "react-i18next";

import Button from "../commonComponents/Button/Button";
import { displayFullName } from "../utils";
import "./TestResultPrintModal.scss";
import logo from "../../img/simplereport-logo-black.svg";
import { QueryWrapper } from "../commonComponents/QueryWrapper";
import LanguageToggler from "../../patientApp/LanguageToggler";
import { formatDateWithTimeOption } from "../utils/date";

export const testQuery = gql`
  query getTestResultForPrint($id: ID!) {
    testResult(id: $id) {
      dateTested
      result
      correctionStatus
      deviceType {
        name
        model
      }
      patient {
        firstName
        middleName
        lastName
        birthDate
      }
      facility {
        name
        cliaNumber
        phone
        street
        streetTwo
        city
        state
        zipCode
        orderingProvider {
          firstName
          middleName
          lastName
          NPI
        }
      }
    }
  }
`;

interface StaticTestResultModalProps {
  testResultId: string | undefined;
  testResult: any;
  hardcodedPrintDate?: string;
}
export const StaticTestResultModal = ({
  testResultId,
  testResult,
  hardcodedPrintDate,
}: StaticTestResultModalProps) => {
  const { t } = useTranslation();
  const {
    patient,
    facility,
    deviceType,
    correctionStatus,
    result,
    dateTested,
  } = testResult;

  return (
    <div
      className={classnames(
        "sr-test-result-report",
        correctionStatus === "REMOVED" && "sr-removed-result"
      )}
    >
      <header className="display-flex flex-align-end flex-justify margin-bottom-1">
        <h1>{t("testResult.result")}</h1>
        <img alt="SimpleReport logo" src={logo} className="sr-print-logo" />
      </header>
      <main>
        <section className="sr-result-section sr-result-patient-details">
          <h2>{t("testResult.patientDetails")}</h2>
          <ul className="sr-details-list">
            <li>
              <b>{t("testResult.name")}</b>
              <div>
                {displayFullName(
                  patient.firstName,
                  patient.middleName,
                  patient.lastName
                )}
              </div>
            </li>
            <li>
              <b>{t("testResult.dob.dateOfBirth")}</b>
              <div>{formatDateWithTimeOption(patient.birthDate)}</div>
            </li>
          </ul>
        </section>
        <section className="sr-result-section sr-result-facility-details">
          <h2>{t("testResult.testingFacility.details")}</h2>
          <ul className="sr-details-list">
            <li>
              <b>{t("testResult.testingFacility.name")}</b>
              <div>{facility.name}</div>
            </li>
            <li>
              <b>{t("testResult.testingFacility.phone")}</b>
              <div>{facility.phone}</div>
            </li>
            <li>
              <b>{t("testResult.testingFacility.address")}</b>
              <div className="sr-result-facility-details-address">
                <span>{facility.street}</span>
                {facility.streetTwo && <span>{facility.streetTwo}</span>}
                <span>
                  {facility.city}, {facility.state} {facility.zipCode}
                </span>
              </div>
            </li>
            <li>
              <b>{t("testResult.testingFacility.clia")}</b>
              <div>{facility.cliaNumber}</div>
            </li>
            <li>
              <b>{t("testResult.testingFacility.orderingProvider")}</b>
              <div>
                {displayFullName(
                  facility.orderingProvider.firstName,
                  facility.orderingProvider.middleName,
                  facility.orderingProvider.lastName
                )}
              </div>
            </li>
            <li>
              <b>{t("testResult.testingFacility.npi")}</b>
              <div>{facility.orderingProvider.NPI}</div>
            </li>
          </ul>
        </section>
        <section className="sr-result-section sr-result-test-details">
          <h2>{t("testResult.testDetails")}</h2>
          <ul className="sr-details-list">
            <li>
              <b>{t("testResult.specimen")}</b>
              <div>{testResultId}</div>
            </li>
            <li>
              <b>{t("testResult.testName")}</b>
              <div>{deviceType.name}</div>
            </li>
            <li>
              <b>{t("testResult.testDevice")}</b>
              <div>{deviceType.model}</div>
            </li>
            <li>
              <b>{t("testResult.testDate")}</b>
              <div>{formatDateWithTimeOption(dateTested, true)}</div>
            </li>
            <li>
              <b>{t("testResult.testResult")}</b>
              <div>
                <strong>
                  {result === "POSITIVE" && t("constants.testResults.POSITIVE")}
                  {result === "NEGATIVE" && t("constants.testResults.NEGATIVE")}
                  {result === "UNDETERMINED" &&
                    t("constants.testResults.UNDETERMINED")}
                </strong>
              </div>
            </li>
          </ul>
        </section>
        <section className="sr-result-section sr-result-next-steps">
          <h2>{t("testResult.note")}</h2>
          {result !== "POSITIVE" && (
            <>
              <p>{t("testResult.notes.negative.p0")}</p>
              <ul className="sr-multi-column">
                <li>{t("testResult.notes.negative.symptoms.li2")}</li>
                <li>{t("testResult.notes.negative.symptoms.li3")}</li>
                <li>{t("testResult.notes.negative.symptoms.li4")}</li>
                <li>{t("testResult.notes.negative.symptoms.li5")}</li>
                <li>{t("testResult.notes.negative.symptoms.li6")}</li>
                <li>{t("testResult.notes.negative.symptoms.li7")}</li>
                <li>{t("testResult.notes.negative.symptoms.li8")}</li>
                <li>{t("testResult.notes.negative.symptoms.li9")}</li>
                <li>{t("testResult.notes.negative.symptoms.li10")}</li>
              </ul>
              <Trans
                t={t}
                parent="p"
                i18nKey="testResult.information"
                components={[
                  <a href="https://www.cdc.gov/coronavirus/2019-ncov/if-you-are-sick/end-home-isolation.html">
                    Centers for Disease Control and Prevention (CDC) website
                  </a>,
                ]}
              />
            </>
          )}
          {result === "POSITIVE" && (
            <>
              <ul>
                <li>{t("testResult.notes.positive.guidelines.li0")}</li>
                <li>{t("testResult.notes.positive.guidelines.li1")}</li>
                <li>{t("testResult.notes.positive.guidelines.li2")}</li>
                <li>{t("testResult.notes.positive.guidelines.li3")}</li>
                <li>{t("testResult.notes.positive.guidelines.li4")}</li>
                <li>{t("testResult.notes.positive.guidelines.li5")}</li>
              </ul>
              <p>{t("testResult.notes.positive.moreInformation")}</p>
              <Trans
                t={t}
                parent="p"
                i18nKey="testResult.notes.positive.p2"
                components={[
                  t("testResult.notes.positive.whenToSeek") +
                    ": " +
                    t("testResult.notes.positive.symptomsLink"),
                ]}
              />
              <ul>
                <li>{t("testResult.notes.positive.emergency.li0")}</li>
                <li>{t("testResult.notes.positive.emergency.li1")}</li>
                <li>{t("testResult.notes.positive.emergency.li2")}</li>
                <li>{t("testResult.notes.positive.emergency.li3")}</li>
                <li>{t("testResult.notes.positive.emergency.li4")}</li>
              </ul>
              <p>{t("testResult.notes.positive.p3")}</p>
              <Trans
                t={t}
                parent="p"
                i18nKey="testResult.notes.positive.difficultNewsLink"
                components={[
                  null,
                  `: ${t("testResult.notes.positive.difficultNewsURL")}`,
                ]}
              />
            </>
          )}
        </section>
      </main>
      <footer>
        <p>
          {t("testResult.printed")}{" "}
          {hardcodedPrintDate || new Date().toLocaleString()}
        </p>
      </footer>
    </div>
  );
};

export interface TestResultPrintModalProps {
  data: any; // testQuery result
  testResultId: string | undefined;
  closeModal: () => void;
  hardcodedPrintDate?: string;
}

export const DetachedTestResultPrintModal = ({
  testResultId,
  data,
  closeModal,
  hardcodedPrintDate,
}: TestResultPrintModalProps) => {
  const { t } = useTranslation();

  const buttonGroup = (
    <div className="sr-result-print-buttons dont-print">
      <Button
        variant="unstyled"
        label={t("testResult.close")}
        onClick={closeModal}
      />
      <Button label={t("testResult.print")} onClick={() => window.print()} />
    </div>
  );

  return (
    <Modal
      isOpen={true}
      className="sr-test-results-modal-content"
      overlayClassName="sr-test-results-modal-overlay"
      contentLabel="Printable test result"
    >
      <div className="display-flex flex-align-center maxw-tablet grid-container patient-header padding-x-0 dont-print">
        <LanguageToggler />
        {buttonGroup}
      </div>
      <StaticTestResultModal
        testResultId={testResultId}
        testResult={data.testResult}
        hardcodedPrintDate={hardcodedPrintDate}
      />
      {buttonGroup}
    </Modal>
  );
};

const TestResultPrintModal = (
  props: Omit<TestResultPrintModalProps, "data">
) => (
  <QueryWrapper<TestResultPrintModalProps>
    query={testQuery}
    queryOptions={{ variables: { id: props.testResultId } }}
    Component={DetachedTestResultPrintModal}
    componentProps={{ ...props }}
  />
);

export default TestResultPrintModal;
