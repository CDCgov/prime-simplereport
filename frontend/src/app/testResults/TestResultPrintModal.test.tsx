import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import MockDate from "mockdate";
import ReactDOM from "react-dom";

import { DetachedTestResultPrintModal } from "./TestResultPrintModal";

const testResult = {
  dateTested: new Date("2022-01-28T17:56:48.143Z"),
  result: "NEGATIVE",
  correctionStatus: null,
  deviceType: {
    name: "Fake device",
  },
  patient: {
    firstName: "First",
    middleName: "Middle",
    lastName: "Last",
    birthDate: "08/07/1990",
  },
  facility: {
    name: "Facility Name",
    cliaNumber: "12D4567890",
    phone: "6318675309",
    street: "555 Fake St",
    streetTwo: null,
    city: "Raleigh",
    state: "NC",
    zipCode: "27601",
    orderingProvider: {
      firstName: "Ordering",
      middleName: null,
      lastName: "Provider",
      NPI: "fake-npi",
    },
  },
  testPerformed: {
    name: "Name",
    loincCode: "",
  },
};

window.print = jest.fn();

describe("TestResultPrintModal", () => {
  let printSpy: jest.SpyInstance;
  let component: any;

  beforeEach(() => {
    ReactDOM.createPortal = jest.fn((element, _node) => {
      return element;
    }) as any;

    printSpy = jest.spyOn(window, "print");

    MockDate.set("2021/01/01");
    component = render(
      <DetachedTestResultPrintModal
        data={{ testResult }}
        testResultId="id"
        closeModal={() => {}}
      />
    );
  });

  afterEach(() => {
    printSpy.mockRestore();
  });

  it("should render the test result print view", async () => {
    userEvent.click(screen.getAllByText("Print")[1]);

    expect(printSpy).toBeCalled();
  });

  it("should render the test date and test time", () => {
    expect(screen.getByText("01/28/2022 5:56pm")).toBeInTheDocument();
  });

  it("matches screenshot", () => {
    expect(component).toMatchSnapshot();
  });
});
