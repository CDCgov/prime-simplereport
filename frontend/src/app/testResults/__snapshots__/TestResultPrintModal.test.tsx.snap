// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestResultPrintModal matches screenshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    aria-hidden="true"
    class="ReactModal__Body--open"
  >
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div>
      <div
        class="ReactModal__Overlay ReactModal__Overlay--after-open sr-test-results-modal-overlay"
      >
        <div
          aria-label="Printable test result"
          aria-modal="true"
          class="ReactModal__Content ReactModal__Content--after-open sr-test-results-modal-content"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="display-flex flex-align-center maxw-tablet grid-container patient-header padding-x-0 dont-print"
          >
            <button
              class="usa-button usa-button--unstyled"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-globe fa-w-16 margin-right-1"
                data-icon="globe"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 496 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
                  fill="currentColor"
                />
              </svg>
              Espa√±ol
            </button>
            <div
              class="sr-result-print-buttons dont-print"
            >
              <button
                class="usa-button usa-button--unstyled"
                type="button"
              >
                Close
              </button>
              <button
                class="usa-button"
                type="button"
              >
                Print
              </button>
            </div>
          </div>
          <div
            class="sr-test-result-report"
          >
            <header
              class="display-flex flex-align-end flex-justify margin-bottom-1"
            >
              <h1>
                SARS-CoV-2 result
              </h1>
              <img
                alt="SimpleReport logo"
                class="sr-print-logo"
                src="simplereport-logo-black.svg"
              />
            </header>
            <main>
              <section
                class="sr-result-section sr-result-patient-details"
              >
                <h2>
                  Patient details
                </h2>
                <ul
                  class="sr-details-list"
                >
                  <li>
                    <b>
                      Name
                    </b>
                    <div>
                      Last, First Middle
                    </div>
                  </li>
                  <li>
                    <b>
                      Date of birth
                    </b>
                    <div>
                      08/07/1990
                    </div>
                  </li>
                </ul>
              </section>
              <section
                class="sr-result-section sr-result-facility-details"
              >
                <h2>
                  Testing facility details
                </h2>
                <ul
                  class="sr-details-list"
                >
                  <li>
                    <b>
                      Facility name
                    </b>
                    <div>
                      Facility Name
                    </div>
                  </li>
                  <li>
                    <b>
                      Facility phone
                    </b>
                    <div>
                      6318675309
                    </div>
                  </li>
                  <li>
                    <b>
                      Facility address
                    </b>
                    <div
                      class="sr-result-facility-details-address"
                    >
                      <span>
                        555 Fake St
                      </span>
                      <span>
                        Raleigh
                        , 
                        NC
                         
                        27601
                      </span>
                    </div>
                  </li>
                  <li>
                    <b>
                      CLIA number
                    </b>
                    <div>
                      12D4567890
                    </div>
                  </li>
                  <li>
                    <b>
                      Ordering provider
                    </b>
                    <div>
                      Provider, Ordering
                    </div>
                  </li>
                  <li>
                    <b>
                      NPI
                    </b>
                    <div>
                      fake-npi
                    </div>
                  </li>
                </ul>
              </section>
              <section
                class="sr-result-section sr-result-test-details"
              >
                <h2>
                  Test details
                </h2>
                <ul
                  class="sr-details-list"
                >
                  <li>
                    <b>
                      Specimen ID
                    </b>
                    <div>
                      id
                    </div>
                  </li>
                  <li>
                    <b>
                      Test name
                    </b>
                    <div>
                      Fake device
                    </div>
                  </li>
                  <li>
                    <b>
                      Test device
                    </b>
                    <div />
                  </li>
                  <li>
                    <b>
                      Test date
                    </b>
                    <div>
                      01/28/2022 5:56pm
                    </div>
                  </li>
                  <li>
                    <b>
                      Test result
                    </b>
                    <div>
                      <strong>
                        Negative
                      </strong>
                    </div>
                  </li>
                </ul>
              </section>
              <section
                class="sr-result-section sr-result-next-steps"
              >
                <h2>
                  Notes
                </h2>
                <p>
                  Contact your health care provider to decide if additional testing is needed, especially if you experience any of these symptoms:
                </p>
                <ul
                  class="sr-multi-column"
                >
                  <li>
                    Shortness of breath or difficulty breathing
                  </li>
                  <li>
                    Fatigue
                  </li>
                  <li>
                    Muscle or body aches
                  </li>
                  <li>
                    Headache
                  </li>
                  <li>
                    Loss of taste or smell
                  </li>
                  <li>
                    Sore throat
                  </li>
                  <li>
                    Congestion or runny nose
                  </li>
                  <li>
                    Nausea or vomiting
                  </li>
                  <li>
                    Diarrhea
                  </li>
                </ul>
                <p>
                  For more information, please visit the 
                  <a
                    href="https://www.cdc.gov/coronavirus/2019-ncov/if-you-are-sick/end-home-isolation.html"
                  >
                    Centers for Disease Control and Prevention (CDC) website
                  </a>
                   or contact your local health department.
                </p>
              </section>
            </main>
            <footer>
              <p>
                Test result printed
                 
                1/1/2021, 12:00:00 AM
              </p>
            </footer>
          </div>
          <div
            class="sr-result-print-buttons dont-print"
          >
            <button
              class="usa-button usa-button--unstyled"
              type="button"
            >
              Close
            </button>
            <button
              class="usa-button"
              type="button"
            >
              Print
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      data-react-modal-body-trap=""
      style="position: absolute; opacity: 0;"
      tabindex="0"
    />
    <div
      class="ReactModalPortal"
    />
  </body>,
  "container": <div>
    <div
      class="ReactModal__Overlay ReactModal__Overlay--after-open sr-test-results-modal-overlay"
    >
      <div
        aria-label="Printable test result"
        aria-modal="true"
        class="ReactModal__Content ReactModal__Content--after-open sr-test-results-modal-content"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="display-flex flex-align-center maxw-tablet grid-container patient-header padding-x-0 dont-print"
        >
          <button
            class="usa-button usa-button--unstyled"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-globe fa-w-16 margin-right-1"
              data-icon="globe"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 496 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
                fill="currentColor"
              />
            </svg>
            Espa√±ol
          </button>
          <div
            class="sr-result-print-buttons dont-print"
          >
            <button
              class="usa-button usa-button--unstyled"
              type="button"
            >
              Close
            </button>
            <button
              class="usa-button"
              type="button"
            >
              Print
            </button>
          </div>
        </div>
        <div
          class="sr-test-result-report"
        >
          <header
            class="display-flex flex-align-end flex-justify margin-bottom-1"
          >
            <h1>
              SARS-CoV-2 result
            </h1>
            <img
              alt="SimpleReport logo"
              class="sr-print-logo"
              src="simplereport-logo-black.svg"
            />
          </header>
          <main>
            <section
              class="sr-result-section sr-result-patient-details"
            >
              <h2>
                Patient details
              </h2>
              <ul
                class="sr-details-list"
              >
                <li>
                  <b>
                    Name
                  </b>
                  <div>
                    Last, First Middle
                  </div>
                </li>
                <li>
                  <b>
                    Date of birth
                  </b>
                  <div>
                    08/07/1990
                  </div>
                </li>
              </ul>
            </section>
            <section
              class="sr-result-section sr-result-facility-details"
            >
              <h2>
                Testing facility details
              </h2>
              <ul
                class="sr-details-list"
              >
                <li>
                  <b>
                    Facility name
                  </b>
                  <div>
                    Facility Name
                  </div>
                </li>
                <li>
                  <b>
                    Facility phone
                  </b>
                  <div>
                    6318675309
                  </div>
                </li>
                <li>
                  <b>
                    Facility address
                  </b>
                  <div
                    class="sr-result-facility-details-address"
                  >
                    <span>
                      555 Fake St
                    </span>
                    <span>
                      Raleigh
                      , 
                      NC
                       
                      27601
                    </span>
                  </div>
                </li>
                <li>
                  <b>
                    CLIA number
                  </b>
                  <div>
                    12D4567890
                  </div>
                </li>
                <li>
                  <b>
                    Ordering provider
                  </b>
                  <div>
                    Provider, Ordering
                  </div>
                </li>
                <li>
                  <b>
                    NPI
                  </b>
                  <div>
                    fake-npi
                  </div>
                </li>
              </ul>
            </section>
            <section
              class="sr-result-section sr-result-test-details"
            >
              <h2>
                Test details
              </h2>
              <ul
                class="sr-details-list"
              >
                <li>
                  <b>
                    Specimen ID
                  </b>
                  <div>
                    id
                  </div>
                </li>
                <li>
                  <b>
                    Test name
                  </b>
                  <div>
                    Fake device
                  </div>
                </li>
                <li>
                  <b>
                    Test device
                  </b>
                  <div />
                </li>
                <li>
                  <b>
                    Test date
                  </b>
                  <div>
                    01/28/2022 5:56pm
                  </div>
                </li>
                <li>
                  <b>
                    Test result
                  </b>
                  <div>
                    <strong>
                      Negative
                    </strong>
                  </div>
                </li>
              </ul>
            </section>
            <section
              class="sr-result-section sr-result-next-steps"
            >
              <h2>
                Notes
              </h2>
              <p>
                Contact your health care provider to decide if additional testing is needed, especially if you experience any of these symptoms:
              </p>
              <ul
                class="sr-multi-column"
              >
                <li>
                  Shortness of breath or difficulty breathing
                </li>
                <li>
                  Fatigue
                </li>
                <li>
                  Muscle or body aches
                </li>
                <li>
                  Headache
                </li>
                <li>
                  Loss of taste or smell
                </li>
                <li>
                  Sore throat
                </li>
                <li>
                  Congestion or runny nose
                </li>
                <li>
                  Nausea or vomiting
                </li>
                <li>
                  Diarrhea
                </li>
              </ul>
              <p>
                For more information, please visit the 
                <a
                  href="https://www.cdc.gov/coronavirus/2019-ncov/if-you-are-sick/end-home-isolation.html"
                >
                  Centers for Disease Control and Prevention (CDC) website
                </a>
                 or contact your local health department.
              </p>
            </section>
          </main>
          <footer>
            <p>
              Test result printed
               
              1/1/2021, 12:00:00 AM
            </p>
          </footer>
        </div>
        <div
          class="sr-result-print-buttons dont-print"
        >
          <button
            class="usa-button usa-button--unstyled"
            type="button"
          >
            Close
          </button>
          <button
            class="usa-button"
            type="button"
          >
            Print
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
